### DISCUSSION TITLE
Using different file extensions than .C and .h for source and header files

### DISCUSSION THREAD

**Discussion by rks171** [idaholab/moose#22053] [Q&A General] [answered]
I have another code that I want to integrate into MOOSE.  A problem is that it uses different file extensions than ".C" and ".h" for the c++ source and header files.  Aside from renaming all the files and updating the include statements, is there a way to tell the MOOSE build system to look for these different file extensions so that I can just copy the files into the MOOSE application?

**Comment by GiudGiud** [comment, marked as answer]
@permcody @cticenhour

**Reply by cticenhour** [reply]
To my knowledge, it would definitely take a modification of the makefiles (maybe `app.mk` if we're talking integration into an app?), as we look for specific file extensions for C++, C, F90 etc. there to create the list of source files. I don't know of a way off-hand to perform that build without needing a modification.

**Reply by rks171** [reply]
Ok that's pretty much what I gathered from looking through those makefiles.  I just wanted to make sure there wasn't something I was missing.

**Reply by permcody** [reply]
Casey is correct - we just do wildcard searchers in app.mk for specific extensions.
